plugins/pulp_rpm/plugins/migrations/0046_add_is_modular_field.py

```diff
@@ -0,0 +1,128 @@
+
+    :return: parsed NEVR (name, epoch, version, release)
+                          str    int     str      str
+    :rtype: tuple
+    """
+    if name.count("-") < 2:  # release or name is missing
+        msg = _("failed to parse nevr '%s' not a valid nevr") % name
+        _LOGGER.exception(msg)
+        raise ValueError(msg)
+
+    release_dash_pos = name.rfind("-")
+    release = name[release_dash_pos + 1:]
+    name_epoch_version = name[:release_dash_pos]
+    name_dash_pos = name_epoch_version.rfind("-")
+    package_name = name_epoch_version[:name_dash_pos]
+
+    epoch_version = name_epoch_version[name_dash_pos + 1:].split(":")
+    if len(epoch_version) == 1:
+        epoch = 0
+        version = epoch_version[0]
+    elif len(epoch_version) == 2:
+        epoch = int(epoch_version[0])
+        version = epoch_version[1]
+    else:
+        # more than one ':'
+        msg = _("failed to parse nevr '%s' not a valid nevr") % name
+        _LOGGER.exception(msg)
+        raise ValueError(msg)
+
+    return package_name, epoch, version, release
+
+
+def migrate_modulemd_artifacts(modulemd):
+    """
+    Set is_modular flag to True for modulemd's artifacts.
+
+    :param modulemd: module for which artifacts should be updated
+    :type  modulemd: pulp_rpm.plugins.db.models.Modulemd
+    """
+    delta = {'is_modular': True}
+    for artifact in modulemd.get('artifacts'):
+        pkg_nevra = nevra(artifact)
+        search_criteria = {'name': pkg_nevra[0],
+                           'epoch': unicode(pkg_nevra[1]),
+                           'version': pkg_nevra[2],
+                           'release': pkg_nevra[3],
+                           'arch': pkg_nevra[4],
+                           NEW_FIELD: {'$exists': False}}
+
+        for rpm in units_rpm_collection.find(search_criteria, ['_id']):
+            units_rpm_collection.update_one({'_id': rpm['_id']}, {'$set': delta})
+
+
+def migrate(*args, **kwargs):
+    """
+    Add `is_modular` field to RPM collection.
+
+    Set this field appropriately for RPM collection based on available info in modulemd units.
+
+    :param args:   unused
+    :type  args:   list
+    :param kwargs: unused
+    :type  kwargs: dict
+    """
+    total_modulemd_units = units_modulemd_collection.count()
+
+    with utils.MigrationProgressLog('Modulemd artifacts', total_modulemd_units) as migration_log:
+        for modulemd in units_modulemd_collection.find({}, ['artifacts']).batch_size(100):
+            migrate_modulemd_artifacts(modulemd)
+            migration_log.progress()
+
+    total_rpm_units = units_rpm_collection.count({NEW_FIELD: {'$exists': False}})
+
+    if total_rpm_units:
+        with utils.MigrationProgressLog('RPM', total_rpm_units) as migration_log:
+            units_rpm_collection.update_many({NEW_FIELD: {'$exists': False}},
+                                             {'$set': {NEW_FIELD: False}})
+            migration_log.progress(migrated_units=total_rpm_units)
```

Comment:
File "/home/ipanova/pulp_development/pulp_rpm/plugins/pulp_rpm/plugins/migrations/0046_add_is_modular_field.py", line 128, in migrate
    migration_log.progress(migrated_units=total_rpm_units)
TypeError: progress() got an unexpected keyword argument 'migrated_units'

Reply:
@ipanova , test with PR for the core pulp/pulp#3706

Reply:
thanks, the PRs were not related so i could not know